internal/util/dataplane/mode.go

```diff
@@ -1,12 +1,43 @@
 package dataplane
 
+import "fmt"
+
+type DBMode string
+
+const (
+	DBModeOff      DBMode = "off"
+	DBModePostgres DBMode = "postgres"
+)
+
+func NewDBMode(mode string) (DBMode, error) {
+	switch mode {
+	case "", string(DBModeOff):
+		return DBModeOff, nil
+	case string(DBModePostgres):
+		return DBModePostgres, nil
+	}
+	return "", fmt.Errorf("unsupported db mode: %q", mode)
+}
+
 // IsDBLessMode can be used to detect the proxy mode (db or dbless).
-func IsDBLessMode(mode string) bool {
-	return mode == "" || mode == "off"
+func IsDBLessMode(mode DBMode) bool {
+	return mode == "" || mode == DBModeOff
 }
 
 // DBBacked returns true if the gateway is DB backed.
 // reverse of IsDBLessMode for readability.
-func DBBacked(mode string) bool {
+func DBBacked(mode DBMode) bool {
```

Comment:
```suggestion
func IsDBBacked(mode DBMode) bool {
```
to make it consistent with `IsDBLessMode` or drop prefix `Is` from both